### ISSUE TITLE
Please provide Tower an easy way to work with "constructed" plugin.

### ISSUE THREAD

**Issue by zaskan** [ansible/awx#8322]
**SUMMARY**
When running in an Ansible Tower environment, the keyed_groups functionality does not work - groups based on facts are not created because tower is executing the following:

`ansible-inventory -i <inventory file> --list --export ...`

instead of:

`ansible-inventory -i existing_inventory -i <inventory file>`

That is the command needed for the plugin to work. We would like to add the possibility to Tower for this to work.

**ISSUE TYPE**

- Feature RFE

**COMPONENT NAME**
- Ansible Tower dynamic inventories
- Constructed inventory plugin

**ANSIBLE VERSION**
Ansible Tower 3.7.2
ansible 2.9.2                                                                                                                                                     

**CONFIGURATION**
```
CACHE_PLUGIN(/etc/ansible/ansible.cfg) = jsonfile                                                                                                                 
CACHE_PLUGIN_CONNECTION(/etc/ansible/ansible.cfg) = /tmp/factcache                                                                                                
CACHE_PLUGIN_TIMEOUT(/etc/ansible/ansible.cfg) = 300                                                                                                              
DEFAULT_GATHERING(/etc/ansible/ansible.cfg) = smart  
```

**OS / ENVIRONMENT**
Operating System: Red Hat Enterprise Linux
CPE OS Name: cpe:/o:redhat:enterprise_linux:7.8:GA:server
Kernel: Linux 3.10.0-1127.13.1.el7.x86_64

**STEPS TO REPRODUCE**
```
plugin: constructed
strict: False

groups:
  ###########################################################################
  # ENVIRONMENT
  ###########################################################################
  env_Production: inventory_hostname.startswith('wsp')
  env_DMZ: inventory_hostname.startswith('wse')
  env_Acceptance: inventory_hostname.startswith('wsa')
  env_Test: inventory_hostname.startswith('wst')
  env_Development: inventory_hostname.startswith('wsd')
  env_Sandbox: inventory_hostname.startswith('wss')

  

keyed_groups:
  
  - prefix: distro
    key: ansible_distribution
  
  - key: ansible_os_name
    prefix: OS

  - key: ansible_os_family
    prefix: os_family

  - key: ansible_windows_domain
    prefix: Domain

```
**EXPECTED RESULTS**
Groups are created based on facts (when run in Tower)

**ACTUAL RESULTS**
No groups created for keyed_groups (when run in Tower)

**_Customer case_**

**Comment by AlanCoding** [comment]
There's a big design discussion here, but all points in this issue should be solved with #3769

> ansible-inventory -i existing_inventory -i \<inventory file\>

That's an over-simplification. You mention Ansible facts - for that to work, facts need to be set up before the update runs. We need to hit this line to write the facts, and inventory updates currently do not do that:

https://github.com/ansible/awx/blob/ba4ae7c1045efc9ba269f6fe8ba5689f71d516a4/awx/main/models/jobs.py#L843

We have other obligations that complicate the matter. We track what inventory _source_ hosts and groups come from, and Ansible itself does no such thing. If an update used `-i existing_inventory -i <inventory file>`, then handling deleted hosts/groups is going to be messy. It's unclear if the criteria for deletion would be that it fails to return a host/group that has the DB relationship to the inventory source, or if we would add novel hostvars to track the source (might be required for multiple inventory files).

**Comment by jakemcdermott** [comment]
:wave: Hello! Since the use case is, apparently, covered by #3769 I'm closing as a duplicate.
